File: spec/python/test_expr_3.py
import unittest

from expr_3 import Expr3

class TestExpr3(unittest.TestCase):
    def test_expr_3(self):
        r = Expr3.from_file("src/fixed_struct.bin")

        self.assertEquals(r.one, 80)
        self.assertEquals(r.two, "ACK")

        self.assertEquals(r.three, "@ACK")
        self.assertEquals(r.four, "_ACK_")
        self.assertEquals(r.is_str_eq, True)
        self.assertEquals(r.is_str_ne, False)
        self.assertEquals(r.is_str_lt, True)
        self.assertEquals(r.is_str_gt, False)
        self.assertEquals(r.is_str_le, True)
        self.assertEquals(r.is_str_ge, False)
        self.assertEquals(r.is_str_lt2, True)

File: spec/javascript/test_expr_3.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('Expr3', 'src/fixed_struct.bin', function(r) {
  assert.equal(r.one, 80);
  assert.equal(r.two, 'ACK');

  assert.equal(r.three, '@ACK');
  assert.equal(r.four, '_ACK_');
  assert.equal(r.isStrEq, true);
  assert.equal(r.isStrNe, false);
  assert.equal(r.isStrLt, true);
  assert.equal(r.isStrGt, false);
  assert.equal(r.isStrLe, true);
  assert.equal(r.isStrGe, false);
  assert.equal(r.isStrLt2, true);
  assert.equal(r.testNot, true);
});
